libs/basics/src/result.ts

```diff
@@ -215,3 +232,87 @@ export type Result<T, E> = Ok<T> | Err<E>;
 export function isResult(value: unknown): value is Result<unknown, unknown> {
   return value instanceof Ok || value instanceof Err;
 }
+
+class ResultBlockError<E> {
+  constructor(private readonly error: E) {}
+
+  err(): E {
+    return this.error;
+  }
+}
+
+/**
+ * Provides an early-return mechanism for functions that return `Result`s,
+ * similar to Rust's `?` operator. This is useful for avoiding `if` statements
+ * that check for errors and return early.
+ *
+ * Note that errors thrown are not caught by this function, so you should use
+ * standard try/catch blocks to catch thrown errors.
+ *
+ * @example
+ *
+ * ```ts
+ * function doSomething(): Result<string, Error> {
+ *   return resultBlock((ret) => {
+ *     const value = doSomethingThatMightFail().or(ret);
+ *     const value2 = doSomethingElseThatMightFail(value).or(ret);
+ *     return anotherThingThatMightFail(value2);
+ *   });
+ * }
```

Comment:
I think it might be worth thinking about making `or` work outside of a `resultBlock` so that we can have a better mental model of what it does. An idea: rename to `okOrElse` or `unwrapOrElse` and let the user provide a default value or function that returns a default value.

I would also suggest maybe changing the naming convention for `ret` to something a bit more descriptive (maybe `bail` or `fail`).

Putting those together:

```ts
return resultBlock((fail) => {
  const value = doSomethingThatMightFail().okOrElse(fail);
  const value2 = doSomethingElseThatMightFail(value).okOrElse(fail);
  return anotherThingThatMightFail(value2);
});
```

Reply:
Yeah, I was trying to keep it short, but I think you're right.

Reply:
Okay, this is now changed.